frontend/lib/src/components/core/Block/utils.ts

```diff
@@ -36,26 +36,36 @@ export function shouldComponentBeEnabled(
 export function isElementStale(
   node: AppNode,
   scriptRunState: ScriptRunState,
-  scriptRunId: string
+  scriptRunId: string,
+  currentFragmentId?: string
 ): boolean {
   if (scriptRunState === ScriptRunState.RERUN_REQUESTED) {
     // If a rerun was just requested, all of our current elements
     // are about to become stale.
     return true
   }
+
   if (scriptRunState === ScriptRunState.RUNNING) {
+    if (currentFragmentId) {
+      return node.fragmentId === currentFragmentId
+    }
     return node.scriptRunId !== scriptRunId
```

Comment:
Can runs with different ids have the same fragment id or is it globally unique? It looks like its the latter, but want to double-check 🙂

Reply:
A `scriptRunId` should be globally unique, but `fragmentId`s are stable across script runs given they're deterministic based on a fragment's module name/qualified function name/and where in the app the fragment is writing to.